delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/IndexingAction.scala

```diff
@@ -36,62 +37,33 @@ trait IndexingAction {
     * @param elem
     *   the element to index
     */
-  def projections(project: ProjectRef, elem: Elem[GraphResource])(implicit
-      cr: RemoteContextResolution
-  ): ElemStream[CompiledProjection]
-
-  /**
-    * Perform an indexing action based on the indexing parameter.
-    *
-    * @param project
-    *   the project in which the resource is located
-    * @param res
-    *   the resource to perform the indexing action for
-    * @param indexingMode
-    *   the execution type
-    */
-  def apply[A](project: ProjectRef, res: ResourceF[A], indexingMode: IndexingMode)(implicit
-      shift: ResourceShift[_, A, _],
-      cr: RemoteContextResolution
-  ): UIO[Unit] = {
-    indexingMode match {
-      case Async => UIO.unit
-      case Sync  =>
-        for {
-          _      <- UIO.delay(logger.debug("Synchronous indexing of resource '{}/{}' has been requested.", project, res.id))
-          // We create the GraphResource wrapped in an `Elem`
-          elem   <- shift.toGraphResourceElem(project, res)
-          errors <- apply(project, elem)
-          _      <- IO.raiseWhen(errors.nonEmpty)(IndexingFailed(res.void, errors.map(_.throwable)))
-        } yield ()
-    }
-  }.hideErrors
+  def projections(project: ProjectRef, elem: Elem[GraphResource]): ElemStream[CompiledProjection]
 
   def apply(project: ProjectRef, elem: Elem[GraphResource])(implicit
-      cr: RemoteContextResolution
-  ): Task[List[FailedElem]] = {
+      contextShift: ContextShift[IO]
+  ): IO[List[FailedElem]] = {
     for {
       // To collect the errors
-      errorsRef <- Ref.of[Task, List[FailedElem]](List.empty)
+      errorsRef <- Ref.of[IO, List[FailedElem]](List.empty)
       // We build and start the projections where the resource will apply
       _         <- projections(project, elem)
+                     .translate(taskToIoK)
                      // TODO make this configurable
                      .parEvalMap(5) {
```

Comment:
I could narrow the problem to this line, it looks like executing in parallel with Monix on Java 17 was the issue ?